apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -2742,15 +2822,22 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
     :ok
   end
 
-  defp send_consumer_properties_payload(realm, device_id, abs_paths_list) do
+  defp send_consumer_properties_payload(realm, device_id, abs_paths_list, compression_format) do
     topic = "#{realm}/#{Device.encode_device_id(device_id)}/control/consumer/properties"
 
     uncompressed_payload = Enum.join(abs_paths_list, ";")
 
-    payload_size = byte_size(uncompressed_payload)
-    compressed_payload = :zlib.compress(uncompressed_payload)
+    payload =
+      case compression_format do
+        :zlib ->
+          payload_size = byte_size(uncompressed_payload)
+          compressed_payload = :zlib.compress(uncompressed_payload)
 
-    payload = <<payload_size::unsigned-big-integer-size(32), compressed_payload::binary>>
+          <<payload_size::unsigned-big-integer-size(32), compressed_payload::binary>>
+
+        :plaintext ->
+          <<255, 255, 255, 255>> <> uncompressed_payload
```

Comment:
See the above comment: the device is now expected to receive a payload according to its purge properties format capability, so there is no more need to prefix the payload with a meaningful header for plaintext